extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -773,78 +773,94 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
+					width: 60,
+					isReadOnly: true,
+					displayName: localizedConstants.FilesText,
+					headerCssStyles: { ...tableHeader, 'text-align': 'left' },
+				},
+				{
+					valueType: azdata.DeclarativeDataType.boolean,
+					width: 80,
+					displayName: localizedConstants.ReadOnlyText,
+					isReadOnly: false,
+					headerCssStyles: { ...tableHeader, 'text-align': 'center' },
+				},
+				{
+					valueType: azdata.DeclarativeDataType.boolean,
+					width: 80,
+					displayName: localizedConstants.DefaultText,
+					isReadOnly: false,
+					headerCssStyles: { ...tableHeader, 'text-align': 'center' },
+				},
+				{
+					valueType: azdata.DeclarativeDataType.boolean,
+					width: 110,
+					displayName: localizedConstants.AutogrowAllFilesText,
+					isReadOnly: false,
+					headerCssStyles: { ...tableHeader, 'text-align': 'center' },
+				}
+			],
 			width: DefaultTableWidth,
-			forceFitColumns: azdata.ColumnSizingMode.DataFit,
+			height: getTableHeight(data.length, DefaultMinTableRowCount, DefaultMaxTableRowCount),
+			dataValues: data,
 			CSSStyles: {
 				'margin-left': '10px'
 			}
 		}).component();
-		this.rowsFilegroupNameContainer = await this.getFilegroupNameGroup(this.rowsFilegroupsTable, FileGroupType.RowsFileGroup);
+		this.rowsFilegroupNameContainer = await this.getFilegroupNameGroupDeclarative(this.rowsFilegroupsTable, FileGroupType.RowsFileGroup);
 		const addButtonComponent: DialogButton = {
 			buttonAriaLabel: localizedConstants.AddFilegroupText,
-			buttonHandler: () => this.onAddDatabaseFileGroupsButtonClicked(this.rowsFilegroupsTable)
+			buttonHandler: () => this.onAddDatabaseFileGroupsButtonClickedDeclarative(this.rowsFilegroupsTable)
 		};
 		const removeButtonComponent: DialogButton = {
 			buttonAriaLabel: localizedConstants.RemoveButton,
-			buttonHandler: () => this.onRemoveDatabaseFileGroupsButtonClicked(this.rowsFilegroupsTable)
+			buttonHandler: () => this.onRemoveDatabaseFileGroupsButtonClickedDeclarative(this.rowsFilegroupsTable)
 		};
-		this.rowsFileGroupButtonContainer = this.addButtonsForTable(this.rowsFilegroupsTable, addButtonComponent, removeButtonComponent);
+		this.rowsFileGroupButtonContainer = this.addButtonsForDeclarativeTable(this.rowsFilegroupsTable, addButtonComponent, removeButtonComponent);
 		this.disposables.push(
-			this.rowsFilegroupsTable.onCellAction(async (arg: azdata.ICheckboxCellActionEventArgs) => {
-				let filegroup = this.rowDataFileGroupsTableRows[arg.row];
-				// Read-Only column
-				if (arg.column === 2) {
-					filegroup.isReadOnly = arg.checked;
-				}
-				// Default column
-				if (arg.column === 3) {
-					this.updateFilegroupsDefaultColumnValues(arg, filegroup, FileGroupType.RowsFileGroup);
-				}
-				// Autogrow all files column
-				if (arg.column === 4) {
-					filegroup.autogrowAllFiles = arg.checked;
-				}
+			this.rowsFilegroupsTable.onDataChanged(async (changedData) => {
+				if (changedData.row && changedData.row > -1) {
```

Comment:
```suggestion
				if (changedData.row > -1) {
```
Greater Than always returns false if one of the arguments is undefined